src/fhirtypes/common.ts

```diff
@@ -301,8 +321,18 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
   } else if (value instanceof FshCode) {
     const [system, ...versionParts] = value.system?.split('|') ?? [];
     const version = versionParts.join('|');
-    const codeSystem = tank.fish(system, Type.CodeSystem);
-    const codeSystemMeta = fisher.fishForMetadata(codeSystem?.name, Type.CodeSystem);
+    let codeSystem = tank.fish(system, Type.CodeSystem);
+    if (!codeSystem) {
+      const csInstance = tank.fish(system, Type.Instance);
+      if (csInstance && isInstance(csInstance) && csInstance.instanceOf === 'CodeSystem') {
+        codeSystem = csInstance;
+      }
+    }
+
+    const codeSystemMeta =
+      codeSystem instanceof FshCodeSystem
+        ? fisher.fishForMetadata(codeSystem?.name, Type.CodeSystem)
+        : fisher.fishForMetadata(codeSystem?.name, Type.Instance);
```

Comment:
If `codeSystem` is null at this point, then here, we just fish for nothing. It might be possible to restructure this so that right after determining that `codeSystem` is something we can use, we immediately fish for metadata in a situation where we know what type to fish for. Something like:
```ts
let codeSystemMeta: Metadata;
if (codeSystem) {
  codeSystemMeta = fisher.fishForMetadata(codeSystem.name, Type.CodeSystem;
} else {
  const csInstance = tank.fish(system, Type.Instance);
  if (csInstance && isInstance(csInstance) && csInstance.instanceOf === 'CodeSystem') {
    codeSystem = csInstance;
    codeSystemMeta = fisher.fishForMetadata(codeSystem.name, Type.Instance);
  }
}
```

Reply:
This is a great point, especially in light of our recent performance gains. Commit `2f25964` takes this suggestion, and also added in a test case ensuring that non-CodeSytem instances aren't being resolved as system urls in concepts.

Reply:
Now that the fishing has been updated to account for instances of CodeSystem, I think it's no longer necessary to have different logic for fishing up the metadata, since `fishForMetadata` calls `fish`. So, I think this can safely be turned back to its original form.

Reply:
You're right! Commit `14bc28dc` returns that bit back to its original form.